cognite/client/utils/_auxiliary.py

```diff
@@ -51,7 +51,10 @@ def get_accepted_params(cls: type[T_CogniteResource]) -> dict[str, str]:
 def fast_dict_load(
     cls: type[T_CogniteResource], item: dict[str, Any], cognite_client: CogniteClient | None
 ) -> T_CogniteResource:
-    instance = cls(cognite_client=cognite_client)
+    try:
+        instance = cls(cognite_client=cognite_client)
+    except TypeError:
+        instance = cls()
```

Comment:
We have `CogniteResource` subclasses that don't accept `cognite_client`? 🤔 Or is the type annot. in need of an update?

Reply:
I don't like that all `CogniteResources` are enforced to have the Cognite client as a paramater in the init method. Very often it is not used, and thus is unnecessary. So this is a suggestion from me.

Reply:
Same, I hate having that client reference everywhere... I wish we could use swap all classes not needing it to `CogniteResponse`... but maybe it is a bit too large of a breaking change... unsure...

Reply:
Keep it here, then maybe I'll add a PR from release/v7 after this is merged